packages/router/src/router.tsx

```diff
@@ -42,57 +124,54 @@ interface PrivateProps {
  * When a user is not authenticated and attempts to visit this route they will be
  * redirected to `unauthenticated` route.
  */
-const Private: React.FC<PrivateProps> = () => {
-  return null
-}
-
-interface PrivatePageLoaderProps {
-  useAuth: typeof useAuthHook
-  unauthenticatedRoute: (args?: Record<string, string>) => string
-  role: string | string[]
-  whileLoading?: () => ReactElement | null
-}
-
-const PrivatePageLoader: React.FC<PrivatePageLoaderProps> = ({
-  useAuth,
-  unauthenticatedRoute,
-  role,
-  whileLoading = () => null,
+const Private: React.FC<PrivateProps> = ({
   children,
+  unauthenticated,
+  role,
 }) => {
-  const { loading, isAuthenticated, hasRole } = useAuth()
+  const routerState = useRouterState()
+  const { isAuthenticated, hasRole } = routerState.useAuth()
 
-  if (loading) {
```

Comment:
While loading has gone missing!

Reply:
No, it has just moved :) 

https://github.com/redwoodjs/redwood/pull/1898/files/d1873abaa85fd672455cf05411a7c7a32f5deeb1#diff-aaaf969c8c446a8f428b54b2d5482767b6d92c7ab4771e68b14145cdbe6a9dddR93-R95

Reply:
But now that you mention it, maybe that won't work. There are some failing tests regarding `<Private>` and auth. I'll keep this in mind when looking at those.

Reply:
All tests are now passing, including the existing test we already had for `whileLoading`. Can you have another look and see if it looks alright to you?